### PR TITLE
feat: list alpaca vault with bCake wrapper

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

### Detailed summary:
- Added `positionManagerVeBCakeWrapper` to the `abi` directory.
- Added `rewardAmount` field to the `OnChainActionResponse` interface.
- Added `bCakeWrapperAddress` and `minDepositUSD` fields to the `PCSDuoTokenVault` interface.
- Added `isThirdPartyVaultConfig` function in `vaults/index.ts`.
- Added `getPositionManagerBCakeWrapperContract` function in `useContract.ts`.
- Added `SAVINGS` manager and related changes in `managers.ts` and `bsc.ts`.
- Added `SAVINGS` strategy and related changes in `utils/strategy.ts` and `ManagerInfo.tsx`.
- Added `bCakeWrapper` and `minDepositUSD` props to `DuoTokenVaultCard` and `LiquidityManagement` components.
- Updated translations in `translations.json`.

### Notable changes:
- Added a new wrapper contract for `positionManagerVeBCakeWrapper`.
- Added a new field `rewardAmount` to the `OnChainActionResponse` interface.
- Added new fields `bCakeWrapperAddress` and `minDepositUSD` to the `PCSDuoTokenVault` interface.
- Added a new function `isThirdPartyVaultConfig` to check if a vault is a third-party vault.
- Added a new function `getPositionManagerBCakeWrapperContract` to get the contract for `positionManagerVeBCakeWrapper`.
- Added a new manager `SAVINGS` and related changes in various files.
- Updated translations in `translations.json`.

> The following files were skipped due to too many changes: `apps/web/src/views/PositionManagers/components/LiquidityManagement.tsx`, `apps/web/src/views/PositionManagers/hooks/useApr.ts`, `apps/web/src/views/PositionManagers/components/RewardAssets.tsx`, `apps/web/src/views/PositionManagers/components/RemoveLiquidity.tsx`, `apps/web/src/views/PositionManagers/components/AddLiquidity.tsx`, `packages/position-managers/src/abi/positionManagerVeBCakeWrapper.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/utils/contractHelpers.ts [hunk 1]
```diff
@@ -277,6 +281,19 @@ export const getPositionManagerWrapperContract = (address: `0x${string}`, signer
   })
 }
 
+export const getPositionManagerBCakeWrapperContract = (
+  address: `0x${string}`,
```
Comment (ChefMomota) [inline apps/web/src/utils/contractHelpers.ts:285]: Address

File: apps/web/src/views/PositionManagers/components/AddLiquidity.tsx [hunk 2]
```diff
@@ -72,6 +72,8 @@ interface Props {
   learnMoreAboutUrl?: string
   lpTokenDecimals?: number
   aprTimeWindow?: number
+  bCakeWrapper?: Address
```
Comment (ChefMomota) [inline apps/web/src/views/PositionManagers/components/AddLiquidity.tsx:75]: How about rename bCakeWrapper -> `bCakeWrapperAddress`

File: packages/position-managers/src/constants/vaults/bsc.ts [hunk 3]
```diff
@@ -4,6 +4,28 @@ import { Strategy, VaultConfig } from '../../types'
...
+    strategy: Strategy.SAVINGS,
+    manager: MANAGER.ALPACA,
+    isSingleDepositToken: true,
+    allowDepositToken0: false,
+    allowDepositToken1: true,
+    managerInfoUrl: 'https://www.alpacafinance.org/',
+    strategyInfoUrl: 'https://docs.alpacafinance.org/leveraged-yield-farming/strategies',
+    learnMoreAboutUrl: 'https://docs.alpacafinance.org/leveraged-yield-farming/strategies/pancakeswap-farms',
+    priceFromV3FarmPid: 8,
+    minDepositUSD: 50,
```
Comment (ChefMomota) [inline packages/position-managers/src/constants/vaults/bsc.ts:27]: Anyway, to confirm `minDepositUSD` amount?

File: apps/web/src/views/PositionManagers/components/AddLiquidity.tsx [hunk 4]
```diff
@@ -351,6 +385,18 @@ export const AddLiquidity = memo(function AddLiquidity({
...
+                  <Text color="text">{t('Your deposit value in USD')}:</Text>
+                  <Text color="text">{`${userTotalDepositUSD.toFixed(2)}`}</Text>
+                </RowBetween>
+              )}
+              {minDepositUSD && (
+                <RowBetween>
+                  <Text color="text">{t('Min deposit USD')}:</Text>
+                  <Text color="text">{`${minDepositUSD}`}</Text>
+                </RowBetween>
+              )}
```
Comment (ChefMomota) [inline apps/web/src/views/PositionManagers/components/AddLiquidity.tsx:399]: Try to combine both

Comment (Chef-Yogi) [reply apps/web/src/views/PositionManagers/components/AddLiquidity.tsx:399]: Done

File: apps/web/src/views/PositionManagers/components/AddLiquidity.tsx (hunk 2 above)
Comment (Chef-Yogi) [reply apps/web/src/views/PositionManagers/components/AddLiquidity.tsx:75]: sure

File: packages/position-managers/src/constants/vaults/bsc.ts (hunk 3 above)
Comment (Chef-Yogi) [reply packages/position-managers/src/constants/vaults/bsc.ts:27]: yes, for this vault, there's a min deposit usd limit

Comment (ChefMomota) [reply packages/position-managers/src/constants/vaults/bsc.ts:27]: I mean this `50`, can I find in which contract?

Comment (Chef-Yogi) [reply packages/position-managers/src/constants/vaults/bsc.ts:49]: let me check

Comment (Chef-Yogi) [reply packages/position-managers/src/constants/vaults/bsc.ts:49]: CleanShot 2024-02-01 at 16 19 47@2x
need to read from another contract 
will add a test case for it

**Review by chefjackson — APPROVED** [review]

**Pull request merged**